### DISCUSSION TITLE
Do I not have to give a api key?

### DISCUSSION THREAD

**Discussion by kodahhhhh** [pieces-app/support#311] [General]
I never gave a key for OpenAI or Claude, but it still works? Is Pieces just fully free for now?

**Comment by hra42** [comment]
Hey @HeyitsRadinn 

Yes, it is! Pieces is VC funded, so they care about growing the product. They plan to introduce a paid subscription later.

If you'd like, you can add an OpenAI API Key:
![CleanShot 2024-07-18 at 14 37 59](https://github.com/user-attachments/assets/89715e6f-c312-4a75-b2db-2da3a20d2491)
![CleanShot 2024-07-18 at 14 38 53](https://github.com/user-attachments/assets/5b0d8f81-0a61-4b92-8fb7-de6ea9f9b603)
![CleanShot 2024-07-18 at 14 40 06](https://github.com/user-attachments/assets/c1dc13c4-83da-499d-93ea-cc42184c4433)

You can take a look at the pricing draft: #104 
For questions like that, you could also join the official pieces discord server:
https://discord.gg/getpieces
